packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -80,12 +80,59 @@ export 'package:flutter/services.dart' show
 
 // BIDIRECTIONAL TEXT SUPPORT
 
+/// An [InheritedElement] that has hundreds of dependencies but will
+/// infrequently change.  This provides a performance tradeoff where building
+/// the [Widget]s is faster but performing updates is slower.
+class _UbiquitousInheritedElement extends InheritedElement {
+  /// Creates an element that uses the given widget as its configuration.
+  _UbiquitousInheritedElement(InheritedWidget widget) : super(widget);
+
+  @override
+  void setDependencies(Element dependent, Object? value) {
+    if (value != null) {
+      throw Exception('Setting aspects on _UbiquitousInheritedElement is not supported.');
+    }
```

Comment:
could just be an assert?

Reply:
I thought about it but I don't know what our policy is there.  You probably know better.  We can't guarantee that the assert would always get hit in debug build and not release build.  If `Directionality` ends up being the only user of it it might not be a big deal.  I was just erring on the side of caution.

Reply:
I would just go with the assert here for now.

Reply:
Done.

Reply:
(oops, fixed it, good now)